### DISCUSSION TITLE
locate docker image after build

### DISCUSSION THREAD

**Discussion by alex116** [firecow/gitlab-ci-local#902] [Q&A] [answered]
I'm trying to figure out where my docker image went.  
I have a job that creates a docker image. 
It's using docker:dind as a service. I see it gets created successfully within gitlab-ci-local when the job is run.  
I do not push it to the gitlab repository when $GITLAB_CI = false so that I only build it locally.  
After gitlab-ci-local is done, the image appears to have disappeared.  
What I want to do is run the command:  
`docker run -ti image-that-was-generated-in-gitlab-ci-local-job bash`  successfully.  
I tried to call gitlab-ci-local with the argument `--no-cleanup` but that didn't work.  

I have a second_job job in the CI that wants to use this generated docker image as a service in order to test it.
I tried calling `gitlab-ci-local second_job --needs --no-cleanup` so the docker image would be created first and the testing job would hopefully find it but that failed as well.

Where can I find this created image? Is there a way to do this with the existing functionality or should I open a feature request?

**Comment by firecow** [comment]
That depends on a lot of things

This question would be a lot easier to answer if you showed us in code, config and logs what you are trying to do, instead of explaining it in your own words.

**Reply by alex116** [reply]
Here's a simple test: 
https://github.com/alex116/gitlab-ci-local-test

`gitlab-ci-local build_test_image` should build an ubuntu image and touch test.txt.  

When the process is done, I want to run this new image: 
`docker run myimage:latest` and it doesn't exist.

**Reply by firecow** [reply, marked as answer]
When using `docker:dind` as a service, the built image only lives for a short while during `build_test_image` execution.
After that the service it shutdown, all images built inside the job during that time just disappears into the blue.

You need a docker image registry, that you should push the built image to, in order for you to use it in subsequent jobs.

I sometimes use shell-executor jobs to built docker images, then they will appear under your `docker image ls`

```yml
build_test_image:
  script:
    - docker build -f Dockerfile -t myimage:latest .
    
ola:
  script:
    - docker run myimage:latest
```


One could also mount ones /var/run/docker.sock into the job locally.

**Reply by alex116** [reply]
thanks, that explains it.
